### DISCUSSION TITLE
Cached code chunks able to depend on external files

### DISCUSSION THREAD

**Discussion by karimn** [quarto-dev/quarto-cli#8721] [Feature Requests]
### Description

It would be very helpful if a code chunk is able to use `dependson` to capture the dependency on a file as well as another chunk. This could be helpful when we need to check if a data file has changed or a compiled source file has been updated.

**Comment by mcanouil** [comment]
Could you elaborate? The option you seemed to describe is a `knitr` option and I don't why it would not work.

**Reply by cscheid** [reply]
I think they mean that they would like our project rendering to understand inter-file dependencies and schedule their execution correctly. We want to do it as part of https://github.com/quarto-dev/quarto-cli/issues/6518 but it's not something that will happen in the short term.

**Reply by karimn** [reply]
It would be nice to be able to do something like

\```{r}
#| dependson: my_data.rds

my_data <- read_rds("my_data.rds")
\```

or 

\```{r}
#| dependson: my_model.stan

my_model <- cmdstanr::cmdstan_model("my_model.stan")
\```

**Reply by mcanouil** [reply]
Is it not what `targets` does?
And knitr when using this option and caching?
Note that `dependson` is a `knitr` option used to create code cells dependencies.
So Quarto cannot use that as you describe.

You should really take a look at `targets` which is compatible with Quarto, especially since you seem to focus only on R.

**Reply by karimn** [reply]
`targets` does look very interesting. I looked here https://books.ropensci.org/targets/literate-programming.html but it doesn't seem to be aware of dependencies between chunks. Are there any plans to incorporate them more closely?

**Reply by mcanouil** [reply]
The whole purpose of targets is to deal with dependencies between "target" or "chunk" in literate programming.
There is even a figure showing that in the link you shared.

**Reply by jaredlander** [reply]
You can achieve this using the `cache.extra` chunk option.

````
```{r}
#| cache.extra: !expr file.mtime("my_model.stan")

my_model <- cmdstanr::cmdstan_model("my_model.stan")
```
````
